napari/tests/util.py

```diff
@@ -0,0 +1,89 @@
+import napari
+import numpy as np
+from napari.layers import (
+    Image,
+    Labels,
+    Points,
+    Shapes,
+    Vectors,
+    Surface
+)
+
+layer_test_data = [
+    (Image, np.random.random((10, 15)),  2),
+    (Image, np.random.random((10, 15, 20)), 3),
+    (Image, [np.random.random(s) for s in [(40, 20), (20, 10), (10, 5)]], 2),
+    (Labels, np.random.randint(20, size=(10, 15)), 2),
+    (Points, 20 * np.random.random((10, 2)), 2),
+    (Vectors, 20 * np.random.random((10, 2, 2)), 2),
+    (Shapes, 20 * np.random.random((10, 4, 2)), 2),
+    (Surface, (np.random.random((10, 3)), np.random.randint(10, size=(6, 3)),
+                 np.random.random(10)), 3)
+]
+
+
+def add_layer_by_type(viewer, layer_type, data):
+    """
+    Convenience method that maps a LayerType to it's add_layer method.
+
+    Parameters
+    ----------
+    layer_type : LayerTypes
+        Layer type to add
+    data :
+        The layer data to add
+    """
+    if layer_type is Labels:
+        return viewer.add_labels(data)
+    if layer_type is Points:
+        return viewer.add_points(data)
+    if layer_type is Vectors:
+        return viewer.add_vectors(data)
+    if layer_type is Shapes:
+        return viewer.add_shapes(data)
+    if layer_type is Surface:
+        return viewer.add_surface(data)
+    if layer_type is Image:
+        return viewer.add_image(data)
+
+
+def view_layer_type(layer_type, data):
+    """
+    Convenience method that maps a LayerType to it's view method.
+
+    Parameters
+    ----------
+    layer_type : LayerTypes
+        Layer type to view
+    data :
+        The layer data to view
+    """
+    if layer_type is Image:
+        return napari.view_image(data)
+    if layer_type is Labels:
+        return napari.view_labels(data)
+    if layer_type is Points:
+        return napari.view_points(data)
+    if layer_type is Vectors:
+        return napari.view_vectors(data)
+    if layer_type is Shapes:
+        return napari.view_shapes(data)
+    if layer_type is Surface:
+        return napari.view_surface(data)
+
+
+def check_viewer_functioning(viewer, view, data, ndim):
+    viewer.dims.ndisplay = 2
+    assert np.all(viewer.layers[0].data == data)
+    assert len(viewer.layers) == 1
+    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
+
+    assert viewer.dims.ndim == ndim
+    assert view.dims.nsliders == viewer.dims.ndim
+    assert np.sum(view.dims._displayed_sliders) == ndim - 2
+
```

Comment:
can you also add some dims reordering in here - see https://github.com/napari/napari/blob/6c5bb037e191bce3c05c5bf23f4f143ecc89a240/napari/tests/test_nD.py#L229 for example, code will depend on ndim